webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +95,1093 @@ public WebSiteSessionController createComponentSessionController(
+
+    request.setAttribute("Site", scc.getWebSite(id));
+    request.setAttribute("AllIcons", scc.getAllIcons());
+    request.setAttribute("ListIcons", scc.getIcons(id));
+
+    String recupParam = request.getParameter("RecupParam"); // =null ou oui
+    String complete = null;
+    if (recupParam != null) {// =oui
+      String nom = request.getParameter("Nom");
+      String description = request.getParameter(DESCRIPTION_PARAM);
+      String lapage = request.getParameter("Page");
+      String listeIcones = request.getParameter(ICON_LIST_PARAM);
+
+      type = "design";
+      complete =
+          "&RecupParam=oui&Nom=" + nom + "&Description=" + description + "&Page=" + lapage +
+              "&ListeIcones=" + listeIcones;
+    } else {
+      return "/webSites/jsp/modifDesc.jsp?Id=" + id + FOLDER_PATH_FILTER + currentPath + "&type=" +
+          type;
+    }
 
-            if (listIcons.size() > 0) {
-              scc.associateIcons(id, listIcons);
-            }
-          } else {
-            request.setAttribute("SearchOK", Boolean.FALSE);
-            request.setAttribute("ListeIcones", listeIcones);
-          }
+    String destination = "/webSites/jsp/modifDesc.jsp?Id=" + id;
+    if (complete != null) {
+      destination += complete;
+    }
+    if (currentPath != null) {
+      destination += FOLDER_PATH_FILTER + currentPath;
+    }
+    if (type != null) {
+      destination += "&type=" + type;
+    }
+    return destination;
+  }
 
-          descriptionSite2 = scc.getWebSite(id);
-          scc.setSessionSite(descriptionSite2);
-          request.setAttribute("Site", descriptionSite2);
-        } else if ("addFolder".equals(action)) {
-          // = null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
+  private String moveDownPublication(final WebSiteSessionController scc, final HttpRequest request)
+      throws WebSitesException {
+    String pubId = request.getParameter("Id");
 
-          // ADD FOLDER -------------------------------------------------------------
-          /* Creer le nouveau repertoire */
-          scc.createFolder(currentPath + "/" + name);
+    scc.changePubsOrder(pubId, 1);
 
-          request.setAttribute("Site", scc.getSessionSite());
+    String id = scc.getSessionTopic().getNodePK().getId();
+    FolderDetail webSitesCurrentFolder = scc.getFolder(id);
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute(CURRENT_FOLDER_PARAM, webSitesCurrentFolder);
 
-        } else if ("renameFolder".equals(action)) {
-          // = null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
-          // = changement de noms des fichiers et repertoires
-          String newName = request.getParameter("newName");
+    return "/webSites/jsp/organize.jsp?Action=Search&Id=" + id;
+  }
 
-          // RENAME FOLDER --------------------------------------------
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Publisher" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Admin" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Admin" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Publisher" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)